project-reports/2023/2023-Q2-Hyperledger-Sawtooth.md

```diff
@@ -0,0 +1,46 @@
+---
+layout: default
+title: 2023 Q2 Hyperledger Sawtooth
+parent: 2023
+grand_parent: Project Updates
+---
+
+# Project Health
+
+Sawtooth is going through a transition from Transact being separated and the idea that Sawtooth 2.0 would use https://github.com/Cargill/splinter . The last two monthly calls have looked at merging Transact and some of the libraries Bitwise and Blockchain TP created with the current 1.3.x branch and renaming it "main".  That would allow the 1.x branch to continue as several companies are using it exclusively for their clients(including Taekion & BlockchainTP)as well as not be dependent on a non-Hyperledger project - Splinter. The advantage is having a blockchain that can easily using different consensus mechanisms easily for different workload situations. 
+
+New contributors have offered to step in a help rebuild the continuous test mode to make it easier to work with and faster.
+
+# Questions/Issues for the TOC
+
+None at this time
+ 
+# Releases
+
+There have been no releases since the last report, but the merged main will be a much improved release.
+
+# Overall Activity in the Past Quarter
+
+The number of posts in Discord has been rising and answers have been answered in a regular fashion.  We have made answering question on Discord and email to be answered in a faster fashion.
+
+# Current Plans
+
+The plans right now have three main objectives:
+
+* Merge the current release 1.2 with Transact and other patches maintained outside of Sawtooth to a new core release.
+* Overhaul the continuous testing Infrastrucure
```

Comment:
What is the plan for Sawtooth 2.0? If it new development, do you think there's opportunity for Hyperledger Foundation to help bring in new developers?

Reply:
I believe the point is that Sawtooth 2.x train did not attract new development, meanwhile the 1.x versions had quite a lot of work built on top of it and continuing.  I would also argue that the foundation bringing in help to go counter to the direction that the community is taking is somewhat worse than counter-productive.

Reply:
Thanks @scealiontach for addressing this point. At the beginning of this report I saw mentions of Sawtooth 2.0 but then there were no mentions of it in the plan. For the 1.x version effort, there are ways to engage the community such as meetup events, working sessions, SIG presentations if any work can attract that forum. Looking forward to seeing the project grow. I did not understand later parts of the comment.

Reply:
To clarify, "I believe the point is that Sawtooth 2.x train did not attract new development" should not have been the take-away from the discussion, but rather that if 1.3.x development exceeded currently main branch development, we can change course and get to 2.x via 1.3.x instead. Its a longer path but would be perfectly fine. "meanwhile the 1.x versions had quite a lot of work built on top of it" -- this is not true currently based strictly on contributions, but I think it would be fair to say instead that there is a lot of interest in continuing 1.3.x development theoretically. I'd rather see a single line of development than two, so if substantial/sustained 1.3.x contributions do materialize, then we need make sure our approach is unified so we are working in the same place (thus my proposal).

@arsulegai There is plenty of areas that new developers could contribute, the easiest of which is with the Rust re-write, which means taking existing python and finish re-writing it into Rust. Or removing or updating the dockerfiles so they work with a newer Ubuntu.